src/pudl_archiver/archivers/msha.py

```diff
@@ -0,0 +1,124 @@
+    "personal_health_samples": "https://arlweb.msha.gov/OpenGovernmentData/DataSets/PersonalHealthSamples.zip",
+    "personal_health_samples_definitions": "https://arlweb.msha.gov/OpenGovernmentData/DataSets/Personal_Health_Samples_Definition_File.txt",
+    "quartz_samples": "https://arlweb.msha.gov/OpenGovernmentData/DataSets/QuartzSamples.zip",
+    "quartz_samples_definitions": "https://arlweb.msha.gov/OpenGovernmentData/DataSets/Quartz_Samples_Definition_File.txt",
+    "violations": "https://arlweb.msha.gov/OpenGovernmentData/DataSets/Violations.zip",
+    "violations_definitions": "https://arlweb.msha.gov/OpenGovernmentData/DataSets/violations_Definition_File.txt",
+    "107a": "https://arlweb.msha.gov/OpenGovernmentData/107a/107aOrders.xlsx",
+}
+""" Dictionary of expected MSHA tables/definition files and pathways."""
+
+
+class MshaArchiver(AbstractDatasetArchiver):
+    """MSHA archiver."""
+
+    name = "mshamines"
+
+    async def get_resources(self) -> ArchiveAwaitable:
+        """Download MSHA resources."""
+        link_pattern = re.compile(
+            r"(?:DataSets\/)([a-zA-Z0-7_]+)(.zip|_Definition_File.txt)"
+        )
+
+        """Get main table links."""
+        links = await self.get_hyperlinks(URL_BASE + EXT_BASE, link_pattern)
+        links = [link.split("/")[-1] for link in links]
+        full_links = [URL_BASE + "DataSets/" + link for link in links]
+
+        """Get 107a table link from separate webpage."""
+        link_pattern_107a = re.compile(r"(?:107a\/)([a-zA-Z0-7]+).xlsx")
+        links_107a = await self.get_hyperlinks(URL_107A + EXT_107A, link_pattern_107a)
+        links_107a = [link.split("/")[-1] for link in links_107a]
+        full_links_107a = [URL_BASE + "107a/" + link for link in links_107a]
+
+        full_links += full_links_107a
+        logger.debug(full_links)
+
+        if any(item not in list(set(datasets.values())) for item in full_links):
+            """If a link to a new dataset is found, raise error."""
+            new_links = " ".join([v for v in full_links if v not in datasets.values()])
+            raise ValueError(f"New dataset download links found: {new_links}")
+
+        if not all(item in full_links for item in list(set(datasets.values()))):
+            """If an expected dataset link is missing, raise error."""
+            missing_data = ", ".join(
+                [k for k, v in datasets.items() if v not in full_links]
+            )
+            raise ValueError(
+                f"Expected dataset download links not found for datasets: {missing_data}"
+            )
+
+        for link in full_links:
+            yield self.get_dataset_resource(link)
+
+    async def get_dataset_resource(self, link: str) -> tuple[Path, dict]:
+        """Download zip and .txt files."""
+        filename = list(datasets.keys())[list(datasets.values()).index(link)]
+        dataset = filename.replace("_definitions", "")
+
+        if ".zip" in link:
```

Comment:
Seems unlikely to be a problem given the expected filenames, but we've definitely been bitten by overly broad string matching in other contexts and this change is easy:

```
if link.endswith(".zip"):
if link.endswith(".txt"):
if link.endswith(".xlsx"):
```